flow/layers/layer.h

```diff
@@ -157,26 +157,52 @@ class Layer {
   // draws a checkerboard over the layer if that is enabled in the PaintContext.
   class AutoSaveLayer {
    public:
-    [[nodiscard]] static AutoSaveLayer Create(const PaintContext& paint_context,
-                                              const SkRect& bounds,
-                                              const SkPaint* paint);
-
+    // Indicates which canvas the layer should be saved on.
+    enum SaveMode {
+      // The layer is saved on the internal_nodes_canvas.
+      kInternalNodesCanvas,
+      // The layer is saved on the leaf_nodes_canvas.
+      kLeafNodesCanvas
```

Comment:
nit: please define internal vs leaf node canvas for the next contributor. why is this distinction necessary?

Reply:
may be worth linking to the go link for https://docs.google.com/document/d/1t_offKOgkuVzvB3fqGTfcltovb-yxqV6tbuP1qIBqVI/edit